crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -388,13 +399,28 @@ impl C8YHttpProxyActor {
             .await?;
         debug!(target: self.name(), "File event created with id: {:?}", event_response_id);
 
-        let build_request = |end_point: &C8yEndPoint| -> Result<HttpRequestBuilder, C8YRestError> {
+        let build_request = |end_point: &C8yEndPoint| {
             let binary_upload_event_url =
                 end_point.get_url_for_event_binary_upload(&event_response_id);
-            Ok(HttpRequestBuilder::post(&binary_upload_event_url)
-                .header("Accept", "application/json")
-                .header("Content-Type", "text/plain")
-                .body(file_content.to_string()))
+
+            async {
+                // TODO: Upload the file as a multi-part stream
+                let file_content = tokio::fs::read(&request.file_path)
+                    .await
+                    .map_err(SMCumulocityMapperError::from)?;
+
+                let (content_type, body) = match String::from_utf8(file_content) {
+                    Ok(text) => ("text/plain", hyper::Body::from(text)),
+                    Err(not_text) => ("application/octet-stream", not_text.into_bytes().into()),
```

Comment:
Much better this way. At least we could avoid that `file_content.clone()`. But, we're still reading the entire file contents to memory(we always were)  and we should explore ways to avoid that, in a separate task of course. I'll create one for that. Passing a `File` handle directly as the `Body` seemed pretty straight-forward with `reqwest`, but not that direct with `hyper`. I still found some samples though, as follows:

```
// Use tokio::io::AsyncRead to stream the file content
    let file_stream = tokio::io::AsyncReadExt::read_to_string(file, String::new()).into_stream();

    // Set the request body to the file content stream
    *request.body_mut() = hyper::Body::wrap_stream(file_stream);
```

May be there are even better ways, but all of that can be explored in a dedicated ticket.

Reply:
I would have had a go, but getting the content-type right is the tricky bit (aka a bit with some design decisions that I don't want blocking the rest of the bugfix). I didn't check this myself, but I assume for getting the file to show up directly in the UI without needing to download it, we need to upload it as `text/plain`. At the moment, I get around this by just attempting to parse the entire thing to UTF-8 and setting content-type based on whether that succeeds.

Reply:
Aah okay. Yeah externally figuring out that content length is gonna be tricky. In the worst case, we can read the file line-by-line and validating if it's all valid utf-8, instead of reading everything into memory. And then pass the file handle to the `RequestBuilder` with the appropriate content type.